### ISSUE TITLE
Crash if size-0 dimension is present in `jnp.ones`

### ISSUE THREAD

**Issue by hawkinsp** [google/jax#16418]
### Description

```
In [1]: import jax, jax.numpy as jnp

In [2]: x = jnp.ones((3,4,9,0), dtype=jnp.uint8)
Metal device set to: Apple M1 Pro

systemMemory: 32.00 GB
maxCacheSize: 10.67 GB

Segmentation fault: 11
```

While degenerate, this is legal code, and should work, not crash.

### What jax/jaxlib version are you using?

jax 0.4.11, jaxlib 0.4.10, jax-metal 0.0.2

**Comment by kulinseth** [comment]
@hawkinsp , this is indeed an issue we are tracking and update here when fix is available.

**Comment by kulinseth** [comment]
@hawkinsp , this has been fixed with me jax-metal==0.3.0
